### DISCUSSION TITLE
How to get Uno WASM to work inside Electron shell?

### DISCUSSION THREAD

**Discussion by andrewkdci** [unoplatform/uno#12498] [Q&A] [answered]
I am trying to run our application as a WASM app in an Electron shell on Ubuntu 22.04. When I load up my app, I just see the Uno Platform logo. Nothing else happens. I've done a google search but the only real leads I find is this SO question ([How to electronize a UNO Wasm application](https://stackoverflow.com/questions/66398406/how-to-electronize-a-uno-wasm-application)), and this GitHub feature request ([Electron Support](https://github.com/unoplatform/uno/issues/624)), which seems to indicate that Uno is **not** supported on Electron. All other searches just bring up a comparison of Uno Platform and Electron.

If true, this is potentially devastating to our project and timeline. Therefore I want to make sure I do my due diligence in investigating this before dropping any bombs. So does anyone know whether there is a way to get an Uno WASM app to run in an Electron shell?

Some relevant info:
Ubuntu Desktop 22.04
Node.js version 12.22.9
Npm version 8.5.1
Electron version 25.0.1

**Comment by andrewkdci** [comment]
A colleague of mine found a possible workaround from this link: [Uno fails to load when UserAgent contains Electron](https://github.com/unoplatform/uno/issues/5409). Basically the workaround is to add this to the mainWindow.loadURL call:

```
    mainWindow.loadURL(
        "http://localhost:8080", {
			userAgent: mainWindow.webContents.getUserAgent()
				.replace("Electron", "WebAppWrapper")
		}
```

My colleague says it works on his system, but it does not work on mine. Instead I get these errors:

![image](https://github.com/unoplatform/uno/assets/130710084/6e4f7912-bd45-4a88-9d60-3e6b9bf09f44)

This all still works in Firefox, though, so I'm not sure why Electron is having trouble retrieving these resources.

**Reply by jeromelaban** [reply]
It's been a while since Electron support was requested, though we still have [this specific sample](https://github.com/unoplatform/Uno.Wasm.Bootstrap/tree/main/src/Uno.Wasm.Tests.Electron) that demonstrates how to use electron.  It's been a while since it's been tested though, but adjustments from the requirejs specifics that can vary between versions, yet it may still be working.

Looking at the errors, it feels like the base path provided to requirejs is incorrect somehow (if it is electron's version which is being used).

**Reply by jeromelaban** [reply]
Also, it's worth mentioning that Uno for Figma runs in an electron shell, using the [embedded mode of the bootstrapper](https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/features-embedded.mode.html?q=embed). That this mode currently only works in the latest `8.0.0-dev` builds of `Uno.Wasm.Bootstrap`.

**Comment by andrewkdci** [comment, marked as answer]
I found the solution. In addition to the following change in the JS script that launches the electron shell:

```
    mainWindow.loadURL(
        "http://localhost:8080", {
			userAgent: mainWindow.webContents.getUserAgent()
				.replace("Electron", "WebAppWrapper")
		}
    );
```

I also needed to upgrade my project from .NET 6 to .NET 7.
